### PR TITLE
DOCSP-43235: Initial 1.44.0 Release Note updates

### PR DESCRIPTION
**Pull request by sarah-olson-mongodb**
## DESCRIPTION
Compass 1.44.0 Release Notes

## STAGING
https://deploy-preview-667--docs-compass.netlify.app/release-notes/

## JIRA
https://jira.mongodb.org/browse/DOCSP-43235

## BUILD LOG
See Netlify comment

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -10,6 +10,30 @@ Release Notes
...
+- Adds per-connection proxy settings (:issue:`COMPASS-8142`).
+- Enables proxy support feature flag  (:issue:`COMPASS-8167`).
+
+Bug Fixes:
+
+- Stream import errors to the log file with proper back pressure 
+  (:issue:`COMPASS-7820`).
+- In the bulk update preview, convert array indexes from strings to 
+  numbers (:issue:`COMPASS-8218`).
+- Bump shell-bson-parser to 1.1.2 (:issue:`MONGOSH-1859`)
```
Comment (kennethdyer) [inline source/release-notes.txt:31]: Line needs a terminal period.

Also, note: without the period it seems like `:issue:` eats the closing parenthesis into its title.

File: source/release-notes.txt [hunk 2]
```diff
@@ -10,6 +10,30 @@ Release Notes
...
 
+|compass| 1.44.0
+----------------
+
+*Released September 3, 2024*
+
+New Features:
+
+- Support for users working with data stored on different connections 
+  stored in the Compass connections window. (:issue:`COMPASS-6410`).
```
Comment (kennethdyer) [inline source/release-notes.txt:21]: I think we should probably trim the period before the parenthesis, since we don't use them elsewhere.

**Review by kennethdyer — APPROVED** [review]
Thanks @sarah-olson-mongodb, I left a few comments but otherwise LGTM.

**Comment by sarah-olson-mongodb** [comment]
External review and LGTM by Betsy Button, over slack.

**Pull request merged**